src/components/ContactsAdd.js

```diff
@@ -1,31 +1,164 @@
+        "linkedIn": formState.linkedIn,
+      })
+    }
+    //Pass the URL we want to pass TO
+    fetch("http://localhost:4000/contacts", opts)
+      .then(res => res.json())
+      .then(data => {
+        formState
+        // console.log("posted contacts:", contacts)
+        console.log("posted contacts formState:", formState)
+      })
+    //reset the form. Timer to allow for post request to complete
+    setTimeout(() => {
+      setFormState(initialState);
+    }, 500)
+
+    //GET request to get teh data back with an ID
+    fetch("http://localhost:4000/contacts")
+      .then(res => res.json())
+      .then(data => {
+        setContacts(data)
+      })
+
+    //bring page back to contactlist: if creation is ok
+    navigate("/")
+
+  }
+
+  //change the input from empty string to submitted info
+  const handleChange = (event) => {
+    //set targets to change
+    const name = event.target.name
+    const value = event.target.value
+
+    if (name === "firstName") {
+      setFormState({ ...formState, firstName: value })
+    }
+    if (name === "lastName") {
+      setFormState({ ...formState, lastName: value })
+    }
+    if (name === "street") {
+      setFormState({ ...formState, street: value })
+    }
+    if (name === "city") {
+      setFormState({ ...formState, city: value })
+    }
+    if (name === "email") {
+      setFormState({ ...formState, email: value })
+    }
+    if (name === "linkedIn") {
+      setFormState({ ...formState, linkedIn: value })
+    }
+    if (name === "twitter") {
+      setFormState({ ...formState, twitter: value })
+    }
+  }
 
   return (
-    <form className="form-stack contact-form">
+    <form className="form-stack contact-form" onSubmit={handleSubmit}>
       <h2>Create Contact</h2>
 
       <label htmlFor="firstName">First Name</label>
-      <input id="firstName" name="firstName" type="text" required />
+      <input id="firstName"
+        name="firstName"
+        type="text"
+        value={formState.firstName}
+        required
+        onChange={handleChange} />
 
       <label htmlFor="lastName">Last Name:</label>
-      <input id="lastName" name="lastName" type="text" required/>
+      <input id="lastName"
+        name="lastName"
+        type="text"
+        value={formState.lastName}
+        required
+        onChange={handleChange} />
 
       <label htmlFor="street">Street:</label>
-      <input id="street" name="street" type="text" required/>
+      <input id="street"
+        name="street"
+        type="text"
+        value={formState.street}
+
+        required
+        onChange={handleChange} />
 
       <label htmlFor="city">City:</label>
-      <input id="city" name="city" type="text" required/>
+      <input id="city"
+        name="city"
+        type="text"
+        value={formState.city}
+        required
+        onChange={handleChange} />
+
+      <label htmlFor="email">Email:</label>
+      <input id="email"
+        name="email"
+        type="text"
```

Comment:
Changing type from text to email will make sure that the provided input has to include a @ and .com .co.uk etc.